### PR TITLE
`drasil-code` documentation 1

### PR DESCRIPTION
**Pull request by Ant13731**
Contributes #2477 
Also changed a type name spelling error. It was only used in the one file where it was defined, so it shouldn't change anything.

### CODE REVIEW

File: code/drasil-build/Build/Drasil/Make/MakeString.hs [hunk 1]
```diff
@@ -39,15 +39,15 @@ renderVar f (Os nm _ _ _) = f nm
 renderVar f (Implicit nm) = "\"" ++ f nm ++ "\""
 renderVar f (Free nm) = f nm
 
--- | Constructor for converting a String into a MakeString.
+-- | Constructor for converting a 'String' into a 'MakeString'.
 makeS :: String -> MakeString
 makeS = Mr
 
--- | Constructor for Windows OS variables.
+-- | Constructor for Windows 'Os' variables.
```
Comment (JacquesCarette) [inline code/drasil-build/Build/Drasil/Make/MakeString.hs:46]: No, here OS is short for Operating System, so having both capitalized was correct (here and elsewhere).

File: code/drasil-code/Language/Drasil/Chunk/Code.hs [hunk 2]
```diff
@@ -20,126 +20,161 @@ import qualified Data.Map as Map
 import Text.PrettyPrint.HughesPJ (render)
 
 -- not using lenses for now
+-- | Information that is needed to translate an idea into code.
```
Comment (JacquesCarette) [inline code/drasil-code/Language/Drasil/Chunk/Code.hs:23]: This class is the other way around: "code as an idea". So it says that a `CodeIdea` must include both some code and its name.

**Review by JacquesCarette — CHANGES_REQUESTED** [review]

**Comment by JacquesCarette** [comment]
Note, some of the GOOL files are huge, and not worth fully documenting with Haddock-style comments right now. Other issues are more pressing.

**Comment by Ant13731** [comment]
By GOOL files, do you mean the ones in `drasil-gool`? Should I continue with the `drasil-code` haddock documentation?

**Comment by JacquesCarette** [comment]
Yes and yes.

**Review by JacquesCarette — APPROVED** [review]

**Pull request merged**